Web v5.7 (#70)
imgly/pesdk-docs: _guides/html5/v5_7/guides/umd.md

> ---

Author: @sascha @elv1n I created some basic docs on how to use the `umd` version and it would be good if you can take a look at this file.

    />
    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
    <title>Photoeditor SDK</title>

    <script src="https://unpkg.com/react@16.13.1/umd/react.production.min.js"></script>
    <script src="https://unpkg.com/react-dom@16.13.1/umd/react-dom.production.min.js"></script>
    <script src="https://unpkg.com/react-dom@16.13.1/umd/react-dom-server.browser.production.min.js"></script>
    <script src="https://unpkg.com/styled-components@4.4.1/dist/styled-components.min.js"></script>
    <script src="https://unpkg.com/photoeditorsdk/umd/no-polyfills.js"></script>
    <style>
      body {
        margin: 0;
      }
    </style>
  </head>
  <body>
    <div id="editor" style="width: 100vw; height: 100vh;"></div>
    <script>
      PhotoEditorSDK.PhotoEditorSDKUI.init({
        container: "#editor",
        license: "",
        image:
          "https://images.unsplash.com/photo-1599713061074-9efa95376d3c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=975&q=80",
        assetBaseUrl: "https://unpkg.com/photoeditorsdk/assets",
      });
    </script>
  </body>
</html>
```

> ## Let's go step by step

Reviewer: I think we need to create a repo with all these steps included instead of step by step guide.
People will need to clone or download the repo.
```
repo/
 assets/
 js/
 index.html
```

After the umd example with external dependencies will be rules for
* peer dependencies 
* assets information
* different index.js and no-polyfills

Author: Yes, we should create a demo repo for this but we should also keep the step but step guide. Especially for frameworks like Angular the customer has to touch 3-4 different files and they would need to look into every subfolder to find them.

Reviewer: better to provide terminal commands for angular users
```
npm install ...
```

##### Adding peer dependencies

> PhotoEditor SDK needs following peer dependencies:

Reviewer 2: ```suggestion
PhotoEditor SDK needs the following peer dependencies:
```

1. React >= 16.8.6
2. React DOM >= 16.8.6
3. Styled Components >= 4.4

We would recommend to download these files and add them to your `js` folder instead of providing external links to your customers.

```html
<script src=".js/photoeditorsdk/react.production.min.js"></script>
<script src=".js/photoeditorsdk/react-dom.production.min.js"></script>
<script src=".js/photoeditorsdk/react-dom-server.browser.production.min.js"></script>

> <script src=".js/photoeditorsdk/styled-components.min.js"></script>

Reviewer 2: Is this supposed to be `.js`? Not `./js` or just `js` or something?

```

##### Adding the PhotoEditor SDK

> The files for the PhotoEditor SDK can be copied from our public [github repo](https://github.com/imgly/pesdk-html5-build/umd/) to your application.

Reviewer 2: ```suggestion
The files for the PhotoEditor SDK can be copied from our public [GitHub repo](https://github.com/imgly/pesdk-html5-build/umd/) to your application.
```

> `umd/index.js` will include all necessary polyfills to get the editor running in IE11 and still does require you to copy `no-polyfills.ts` to your project.

Reviewer 2: ```suggestion
`umd/index.js includes all necessary polyfills to get the editor running in IE11. You must also copy `no-polyfills.ts` to your project as this is required by the `index.js` file.
```

Not 100% sure here. Is my correction what you meant to say here? Also, is it `no-polyfills.ts` or `no-polyfills.js`?

```html
<script src="./js/photoeditorsdk/index.js"></script>
```

If you want to provide your own polyfills you will only need `umd/no-polyfills.js`

```html
<script src="./js/photoeditorsdk/no-polyfills.js"></script>
```

##### Initialising the PhotoEditor SDK

```html
<body>
  <div id="editor" style="width: 100vw; height: 100vh;"></div>
  <script>
    PhotoEditorSDK.PhotoEditorSDKUI.init({
      container: "#editor",
      license: "",
      image: "",
      assetBaseUrl: "",
    });
  </script>
</body>
```

##### Providing the assets

> The assets folder can be copied from the public [github repo](https://github.com/imgly/pesdk-html5-build/) to your project.

Reviewer 2: ```suggestion
The assets folder can be copied from the public [GitHub repo](https://github.com/imgly/pesdk-html5-build/) to your project.
```
```suggestion
The assets folder can be copied from the public [github repo](https://github.com/imgly/pesdk-html5-build/) to your project.
```

Afterwards you need to provide the assets path to `assetBaseUrl` e.g.: `assetBaseUrl: "assets/photoeditor"`.

The `assetBaseUrl` will also accept an external link and you are free to provide the assets from an AWS bucket or your own server.

{% capture identifier %}{{page.title}}-{{page.version}}-ANALYTICS-02{% endcapture %}

> {% include multilingual_code_block.html snippets=snippets identifier=identifier %}

Reviewer 2: I don't think you need this here. Afaik this is only used when providing a code block that requires a language switcher (e.g. to switch between Java and Kotlin or Swift and Objective-C)

<!-- <div class="important-notice"> -->
<div class="documentation__disclaimer">
<h4 id="cors">CORS</h4> 
If you are loading images from external sources (e.g. from an AWS bucket), you need to first configure <b>Cross-Origin Resource Sharing</b> for both the server and the image. <br><br>
Otherwise, you will see errors such as <br>
<b><em>Failed to execute 'texImage2D' on 'WebGLRenderingContext': The cross-origin image at [...] may not be loaded.</em></b> <br>
or <br>
<b><em> Unable to get image data from canvas because the canvas has been tainted. </em></b> <br>
<br>
Please follow the instructions on how to properly configure CORS <a href="{{site.baseurl}}/guides/html5/v5_7/faq/cors">here</a>.
</div>

## Ready to go!

There you have it. PhotoEditor SDK for the Web is ready to use. Refer to the [configuration documentaion]({{site.baseurl}}/guides/html5/v5_7/introduction/configuration) for more configuration options.

> {% capture identifier %}{{page.title}}-{{page.version}}-ANALYTICS-03{% endcapture %}

Reviewer 2: Same as above, don't think this is needed.
